controllers/clustertopology_controller_compute.go

```diff
@@ -140,9 +141,129 @@ func (r *ClusterTopologyReconciler) getClass(ctx context.Context, cluster *clust
 	return class, nil
 }
 
-// getTemplate gets the object referenced in ref.
+// getCurrentState gets information about the current state of a Cluster by inspecting the state of the InfrastructureCluster,
+// the ControlPlane, and the MachineDeployments associated with the cluster.
+func (r *ClusterTopologyReconciler) getCurrentState(ctx context.Context, cluster *clusterv1.Cluster, class clusterv1.ClusterClass) (*clusterTopologyState, error) {
+	var clusterState clusterTopologyState
+
+	// Reference to the InfrastructureCluster can be nil and is expected to be on the first reconcile loop
+	// In this case the method should still be allowed to continue.
+	if cluster.Spec.InfrastructureRef != nil {
+		infra, err := r.getCurrentInfrastructureClusterState(ctx, cluster)
```

Comment:
```suggestion
		clusterState.infrastructureCluster, err = r.getCurrentInfrastructureClusterState(ctx, cluster)
```
If you want. This would also apply to the two other cases below

(this also requires `var err error` somewhere at the top)

Reply:
I think I'll leave it as is - I don't see the advantage to this version.

Reply:
It avoids temporary variables we don't need and is more concise :) But it's okay for me if you want to keep it as is.